src/CodeBlock.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { ChatBotState } from './state';
+
+/**
+ * If content is provided, display content. Otherwise display the code stored
+ * in the redux store
+ */
+export const CodeBlock = (props: { content?: string }) => {
+  const code = useSelector<ChatBotState>((state: ChatBotState) => state.code);
```

Comment:
Glaube der Typ vom Argument wird schon durch den generic definiert.

```suggestion
  const code = useSelector<ChatBotState>(state => state.code);
```